lite/valor_lite/detection/manager.py

```diff
@@ -817,45 +908,129 @@ def add_data(
             # cache labels and annotations
             keyed_groundtruths = defaultdict(list)
             keyed_predictions = defaultdict(list)
+
             for gidx, gann in enumerate(detection.groundtruths):
-                self._evaluator.groundtruth_examples[uid_index][
-                    gidx
-                ] = np.array(gann.extrema)
+                if not isinstance(gann, annotation_type):
+                    raise ValueError(
+                        f"Expected {annotation_type}, but annotation is of type {type(gann)}."
+                    )
+
+                if annotation_type == BoundingBox:
+                    self._evaluator.groundtruth_examples[uid_index][
+                        gidx
+                    ] = _get_annotation_representation(obj=gann)
```

Comment:
Could you add in a bbox conversion step for polygon and bitmask (e.g. `Bitmask.to_box()`) so that we can cache bounding box examples for those annotations too?

Reply:
made this change, but note that _add_data now returns examples for polygons / bitmasks while _add_data_from_valor_dict doesn't